562 add more page

davidtrussler opened the pull request:
Trello: [562 | Link More drop down in the sub navigation to new page with "more" links](https://trello.com/c/UCCxgSnI/562-link-more-drop-down-in-the-sub-navigation-to-new-page-with-more-links)

This PR creates a new view which users will be directed to when they click on the "More" link in the navigation. Screenshot below. 

Changes: 
- Updates routes file for the new page
- Adds a new controller for the new page
- Adds a new view for the new page
- Adds a test for the new page
- Adds temporary "forbidden" toggle for users without Design System permissions
- Reorders links and changes "Departments & Agencies" to "Organisations"

![Screenshot 2023-10-25 at 16 27 26](https://github.com/alphagov/whitehall/assets/6080548/a2c89caf-72b9-47de-a3ca-57050515dcf9)

minhngocd commented:
Hi @davidtrussler All looks good to me, just a question about the tests. Should we add a test somewhere that checks if the user is a preview design user they should see the 1 link without the list, and if they are without the permission they should?

davidtrussler commented:
> Hi @davidtrussler All looks good to me, just a question about the tests. Should we add a test somewhere that checks if the user is a preview design user they should see the 1 link without the list, and if they are without the permission they should?

Thinking about the test, and if I understand what you are saying correctly, wouldn't that be a test that would belong in the [Transition sub navigation bar to the new design system](https://trello.com/c/LsOoy6Dv/563-transition-sub-navigation-bar-to-the-new-design-system) card? This page only exists in the design system scenario, but the "More" link appears in both but depends on the user's permission as to what it actually is.

minhngocd commented:
> > Hi @davidtrussler All looks good to me, just a question about the tests. Should we add a test somewhere that checks if the user is a preview design user they should see the 1 link without the list, and if they are without the permission they should?
> 
> Thinking about the test, and if I understand what you are saying correctly, wouldn't that be a test that would belong in the [Transition sub navigation bar to the new design system](https://trello.com/c/LsOoy6Dv/563-transition-sub-navigation-bar-to-the-new-design-system) card? This page only exists in the design system scenario, but the "More" link appears in both but depends on the user's permission as to what it actually is.

Heyya @davidtrussler . 
I think we still need a test for what the content is based on user permission. I can see there is logic for "if user is design system then show 1 link, otherwise show a list of tags" but I don't see a test for this logic.

davidtrussler commented:
> > > Hi @davidtrussler All looks good to me, just a question about the tests. Should we add a test somewhere that checks if the user is a preview design user they should see the 1 link without the list, and if they are without the permission they should?
> > 
> > 
> > Thinking about the test, and if I understand what you are saying correctly, wouldn't that be a test that would belong in the [Transition sub navigation bar to the new design system](https://trello.com/c/LsOoy6Dv/563-transition-sub-navigation-bar-to-the-new-design-system) card? This page only exists in the design system scenario, but the "More" link appears in both but depends on the user's permission as to what it actually is.
> 
> Heyya @davidtrussler . I think we still need a test for what the content is based on user permission. I can see there is logic for "if user is design system then show 1 link, otherwise show a list of tags" but I don't see a test for this logic.

@minhngocd Ah right, I understand. That's in the helper, right? That doesn't quite do that, it just formats the stuff that's returned in a way that is more useable in the (list) component we are using when we're in the design system. I tried to cover that with the existing test (checking for a `govuk-link` element) though there might be a better way. In other words the "more" view will only render in the design system rather than the same view rendering differently depending on permissions.

minhngocd commented:
> > > > Hi @davidtrussler All looks good to me, just a question about the tests. Should we add a test somewhere that checks if the user is a preview design user they should see the 1 link without the list, and if they are without the permission they should?
> > > 
> > > 
> > > Thinking about the test, and if I understand what you are saying correctly, wouldn't that be a test that would belong in the [Transition sub navigation bar to the new design system](https://trello.com/c/LsOoy6Dv/563-transition-sub-navigation-bar-to-the-new-design-system) card? This page only exists in the design system scenario, but the "More" link appears in both but depends on the user's permission as to what it actually is.
> > 
> > 
> > Heyya @davidtrussler . I think we still need a test for what the content is based on user permission. I can see there is logic for "if user is design system then show 1 link, otherwise show a list of tags" but I don't see a test for this logic.
> 
> @minhngocd Ah right, I understand. That's in the helper, right? That doesn't quite do that, it just formats the stuff that's returned in a way that is more useable in the (list) component we are using when we're in the design system. I tried to cover that with the existing test (checking for a `govuk-link` element) though there might be a better way. In other words the "more" view will only render in the design system rather than the same view rendering differently depending on permissions.

@davidtrussler  Ahhh Now that i've dug into what this actually does I get it now. Sorry about that.
I was under the impression we were also going to remove the dropdown for the More on the header itself. I think maybe we can try and shift this branch so that it is based off the Main header transition branch and directly on the new header partial (as opposed to the legacy header), we can remove the dropdown of More. What do you think?

davidtrussler commented:
> > @minhngocd Ah right, I understand. That's in the helper, right? That doesn't quite do that, it just formats the stuff that's returned in a way that is more useable in the (list) component we are using when we're in the design system. I tried to cover that with the existing test (checking for a `govuk-link` element) though there might be a better way. In other words the "more" view will only render in the design system rather than the same view rendering differently depending on permissions.
> 
> @davidtrussler Ahhh Now that i've dug into what this actually does I get it now. Sorry about that. I was under the impression we were also going to remove the dropdown for the More on the header itself. I think maybe we can try and shift this branch so that it is based off the Main header transition branch and directly on the new header partial (as opposed to the legacy header), we can remove the dropdown of More. What do you think?

@minhngocd I spoke with Nik about this and we've moved the bit about removing the dropdown (on both "More" and "New document") to the Acceptance Criteria for the subnav card, and kept the two new pages associated with those just to be adding views. So as it stands that is part of that work rather than this, which I think makes more sense as it's less reliant on work in progress elsewhere.

davidtrussler commented:
I've made a couple of updates to this PR to make some changes discussed: 
- Reordered the links and changed "Departments & Agencies" to "Organisations" @nnagewad 
- Added helper methods to display the links in the dropdown correctly @minhngocd
